src/node_options.h

```diff
@@ -0,0 +1,183 @@
+                                 "!eNULL:"                          \
+                                 "!EXPORT:"                         \
+                                 "!DES:"                            \
+                                 "!RC4:"                            \
+                                 "!MD5:"                            \
+                                 "!PSK:"                            \
+                                 "!SRP:"                            \
+                                 "!CAMELLIA"
+#endif
+
+#define V8_DEFAULT_THREAD_POOL_SIZE 4;
+
+enum NodeOptionFlags {
+  NODE_OPTION_ABORT_ON_UNCAUGHT_EXCEPTION,
+  NODE_OPTION_AT_SECURE,
+  NODE_OPTION_INTERACTIVE,
+  NODE_OPTION_NO_DEPRECATION,
+  NODE_OPTION_EXPERIMENTAL_MODULES,
+  NODE_OPTION_EXPERIMENTAL_VM_MODULES,
+  NODE_OPTION_EXPOSE_INTERNALS,
+  NODE_OPTION_EVAL_STRING,
+  NODE_OPTION_NO_FORCE_ASYNC_HOOKS_CHECKS,
+  NODE_OPTION_NO_WARNINGS,
+#if HAVE_OPENSSL
+  NODE_OPTION_OPENSSL_CERT_STORE,
+# if NODE_FIPS_MODE
+  NODE_OPTION_OPENSSL_ENABLE_FIPS,
+  NODE_OPTION_OPENSSL_FORCE_FIPS,
+# endif
+#endif
+  NODE_OPTION_PENDING_DEPRECATION,
+  NODE_OPTION_PRESERVE_SYMLINKS,
+  NODE_OPTION_PRINT_EVAL,
+  NODE_OPTION_PROF_PROCESS,
+  NODE_OPTION_SYNTAX_CHECK,
+  NODE_OPTION_THROW_DEPRECATION,
+  NODE_OPTION_TRACE_DEPRECATION,
+  NODE_OPTION_TRACE_ENABLED,
+  NODE_OPTION_TRACE_SYNC_IO,
+  NODE_OPTION_TRACE_WARNINGS,
+  NODE_OPTION_TRACK_HEAP_OBJECTS,
+  NODE_OPTION_ZERO_FILL_BUFFERS
+};
+
+// The node::NodeOption class encapsulates startup and options details
+// about the Node.js process. The NodeOption object is created within
+// node_main.cc and passed down to collect current state detail.
+class NodeOption {
+ public:
+  NodeOption();
+  ~NodeOption() {}
+
+  bool SafeGetenv(const char* key, std::string* text);
+
+  int v8_thread_pool_size() const { return v8_thread_pool_size_; }
+  uint64_t prog_start_time() const { return prog_start_time_; }
+  std::string warning_file() const { return warning_file_; }
+  std::string eval_string() const { return eval_string_; }
+  std::string userland_loader() const { return userland_loader_; }
+  std::string trace_enabled_categories() const {
+    return trace_enabled_categories_;
+  }
+  std::vector<std::string> preload_modules() const { return preload_modules_; }
+
+  void SetProgStartTime() {
+    prog_start_time_ = uv_now(uv_default_loop());
+  }
+
+  void Revert(int cve) {
+    reverted_ |= (1 << cve);
+  }
+
+  bool IsReverted(int cve) {
+    return reverted_ & (1 << cve);
+  }
+
+  void set_v8_thread_pool_size(int size) {
+    v8_thread_pool_size_ = size;
+  }
+
+  void set_warning_file(const char* warning_file) {
+    warning_file_ = warning_file;
+  }
+
+  void set_eval_string(const char* eval_string) {
+    Set(NODE_OPTION_EVAL_STRING);
+    eval_string_ = eval_string;
+  }
+
+  void set_userland_loader(const char* loader) {
+    userland_loader_ = loader;
+  }
```

Comment:
I guess it would be nice to have some kind of common handling for all string properties, like for the booleans…

Reply:
Yeah, that's on my list of considerations for a future PR